### PR TITLE
Ports use_tool proc from /tg/

### PR DESCRIPTION
**Pull request by alsoandanswer**
Self-explanatory. As of now, from what I've gleaned, basically rolls together do_after and usesound into one, so you don't have to manually plug in play_sound, do_after, toolspeed, and whatever.

It is supposed to add a user.is_busy() check, so like, no two people can screwdriver the same thing at the same time or whatever, but that's for a future PR.

Might consider replacing all play_sound usesounds, with use_tool of a delay of 0, for code-cleanliness and whatever.

Also, adds hammer and shovel usesounds. Freesound, edited by me, yadda yadda yadda.

Now, why'd I add this? Apparently I need all of this just to add a funny welder animation.

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -871,6 +871,82 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
...
+	// No delay means there is no start message, and no reason to call tool_start_check before use_tool.
+	// Run the start check here so we wouldn't have to call it manually.
+
+	//if(user.is_busy()) to be ported in future if we ever need it
+	//	return
+
+	if(!delay && !tool_start_check(user, amount))
+		return
+
+	delay *= toolspeed
```
Comment (JohnWildkins) [inline code/game/objects/items.dm:884]:
```suggestion
	delay /= toolspeed
```
TG's toolspeed is smaller = better, ours is larger = better, so right now in this PR the impact driver takes 3x as long to do anything

**Review by JohnWildkins — CHANGES_REQUESTED** [review]

**Comment by alsoandanswer** [comment]
!review

**Comment by alsoandanswer** [comment]
!review

**Review by JohnWildkins — APPROVED** [review]

File: code/game/objects/items/weapons/tools.dm [hunk 2]
```diff
@@ -483,12 +483,11 @@
...
 	return REAGENT_VOLUME(reagents, /decl/reagent/fuel)
 
-//Removes fuel from the welding tool. If a mob is passed, it will perform an eyecheck on the mob. This should probably be renamed to use()
-/obj/item/weldingtool/proc/remove_fuel(var/amount = 1, var/mob/M = null, var/colourChange = TRUE)
+//Removes fuel from the welding tool. If a mob is passed, it will perform an eyecheck on the mob.
+/obj/item/weldingtool/use(var/amount = 1, var/mob/M = null, var/colourChange = TRUE)
 	if(!welding)
 		return 0
 	else if(welding > 0 && colourChange)
-		set_light(0.7, 2, l_color = LIGHT_COLOR_CYAN)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/tools.dm:491]: why is this gone?

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 3]
```diff
@@ -969,7 +969,7 @@
 	else if(O.isscrewdriver())
 		if(length(contents) == 0)
 			to_chat(user, SPAN_NOTICE("You begin poking holes in \the [src]."))
-			if (do_after(user, 10/O.toolspeed, act_target = src))
+			if(O.use_tool(src, user, 30))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/storage/boxes.dm:972]: is the removal of the toolspeed division in all of these intentional? also, what's up with increasing the timers? document them

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/tools.dm (hunk 2 above)
Comment (alsoandanswer) [reply code/game/objects/items/weapons/tools.dm:491]: Already set in update_torch proc.

File: code/game/objects/items/weapons/storage/boxes.dm (hunk 3 above)
Comment (alsoandanswer) [reply code/game/objects/items/weapons/storage/boxes.dm:972]: Yes. It's intentional. All toolspeed is handled by the use_tool proc itself, so people don't need to copypasta it everytime.

Comment (alsoandanswer) [reply code/game/objects/items/weapons/storage/boxes.dm:972]: ill add whatever timers i tweaked to the changelog

**Comment by alsoandanswer** [comment]
updated changelog

**Comment by alsoandanswer** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**